src/main/java/bio/terra/service/snapshotbuilder/utils/CriteriaQueryBuilder.java

```diff
@@ -170,4 +170,28 @@ public Query generateRollupCountsQueryForCriteriaGroupsList(
 
     return new Query(List.of(personId), List.of(rootTable), filterVariable);
   }
+
+  public Query generateRowIdQueryForCriteriaGroupsList(
+      List<List<SnapshotBuilderCriteriaGroup>> criteriaGroupsList) {
```

Comment:
I didn't notice this before, but I think this and `generateRollupCountsQueryForCriteriaGroupsList()` should accept `List<SnapshotBuilderCohort>` instead of `List<List<SnapshotBuilderCriteriaGroup>>`.